src/reconcile/rec_visibility.c

```diff
@@ -419,6 +418,21 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
+                /*
+                 * Handle double tombstone scenario that can happen in the history store when we add
+                 * zero timestamped tombstones to the front of the update chain.
+                 */
+                if (upd->next != NULL && upd->next->type == WT_UPDATE_TOMBSTONE) {
+                    WT_ASSERT(session, upd->start_ts == 0 && WT_IS_HS(S2BT(session)));
+                    upd = upd->next;
+                    WT_TIME_WINDOW_SET_STOP(select_tw, upd);
+                    tombstone = upd;
+                    /*
+                     * Pin the 0 timestamped tombstone in cache until it becomes globally visible.
+                     */
+                    has_newer_updates = true;
+                }
+
```

Comment:
We should revise the comment

/* If we only have a tombstone in the update list, we must have an ondisk value. */

We may have two tombstones on the update chain when we go into that code now.